data/processors/nat.py

```diff
@@ -0,0 +1,195 @@
+            "5269",
+            # Former ITEM (Innovationszentrum für Therapeutische Medizintechnik), but placed
+            # exactly where 5701 is now, probably no longer existing.
+            # https://portal.mytum.de/pressestelle/pressemitteilungen/news-500)
+            "5702",
+            # Others
+            # ¹WZW Source: https://wiki.tum.de/download/attachments/1318389295/141016_WZW_Plan_Final.pdf?api=v2
+            "4118",  # "Kindervilla II"; Maybe ID change from 4118 to 4117
+            "4198",  # "ehem. Wasserwerk"; Probably ID change from 4198 to 4192
+            "4228",  # "Trafostation XI"; Maybe ID change from 4228 to 4281
+            "4305",  # "Tierernährung NGB"; Probably now part of 4318¹
+            "4306",  # "Tierernährung HGB"; Probably now part of 4318¹
+            "4312",  # "Zierpflanzenbau Gewächshaus"; Probably ID change from 4311 to 4379¹
+            "4313",  # "Zierpflanzenbau Wirtschaftsgebäude"; Can't find any location
+            "4919",  # "Getreidesilo"; Maybe ID change from 4919 to 4010
+            # Location unknown
+            "2603",  # Tankstelle
+            "4112",  # "Institutsgebäude Brauwesen"; Maybe 48.39508 / 11.72437 (near 4111)
+            "4133",  # Wohngebäude Ganzenmüllerstr.
+            "4280",  # Ehem. Arbeitsamt
+            "4303",  # Vegetationsh. Inst. Pfl. und Züchtung
+            "5216",  # CO 60 Quelle
+            # They might be integrated custom somewhere else, but here we ignore these.
+            "3002",  # "Testgebäude 2" => building which probably does not exist
+            "5110",  # wurde Abgerissen
+            "5537",  # "Interims-Zelt-TUMshop" => building which no longer exists
+            "0598",
+            "4298",
+            "5538",
+            "5998",  # "Interims-Tentomax => buildings no longer exist
+            "5516",
+            "5600",  # phantom buildings, which don't exist
+            "XXXX",  # "virtueller Raum"
+        ]:
+            continue
+
+        _merge_building(data, building)
+
+
+def _infer_internal_id(building, data):
+    # The NAT Roomfinder has buildings in it, that are not in TUMonline
+    # (for example Max-Planck-Institut für Plasmaphysik). We keep them,
+    # but use a different building id.
+    if building.b_code.startswith("X"):
+        if building.b_code == "XUCL":
+            building.b_id = "origins-cluster"
+        else:
+            building.b_id = building.b_code[1:].lower()
+
+        return building.b_id
+
+    for _id, _data in data.items():
+        if "b_prefix" in _data and _data["b_prefix"] == building.b_code:
+            if building.b_id is not None:
+                raise RuntimeError(f"Building id '{building.b_code}' more than once in base data")
+            building.b_id = _id
+    if building.b_id is None:
+        raise RuntimeError(
+            f"Building '{building}' not found in base data. " f"It may be missing in the areatree.",
+        )
+    return building.b_id
```

Comment:
Because it was very inefficient, I've simplified this in #495. Only building 1720 was concerned